sdk/python/kfp/components/pipeline_task.py

```diff
@@ -474,6 +475,28 @@ def my_pipeline():
             self._task_spec.dependent_tasks.append(task.name)
         return self
 
+    def ignore_upstream_failure(self) -> 'PipelineTask':
+        """If called, the pipeline task will run when any specified upstream
+        tasks complete, even if unsuccessful.
+
+        This method effectively turns the caller task into an exit task
+        if the caller task has upstream dependencies.
+        """
+
+        from kfp.components import pipeline_context
+        for input_spec_name, input_spec in self.component_spec.inputs.items():
+            for input_name, argument_value in self._inputs.items():
+                if (input_spec_name == input_name) and (isinstance(
+                        argument_value, pipeline_channel.PipelineChannel)) and (
+                            not input_spec.optional) and (
+                                argument_value.task_name in pipeline_context
+                                .Pipeline.get_default_pipeline().tasks):
+                    raise ValueError(
+                        f'Exit task {self.name} requires a default value to make sure the Exit handler never fails.'
+                    )
+
+        self.ignore_upstream_failure_tag = True
```

Comment:
Maybe prefix with underscore? We don't do this consistently for all other `PipelineTask` attributes, but probably better if we do.